src/types/profile.ts

```diff
@@ -5,23 +5,12 @@ type AnyMapper = {
   mapMany: (sourceKey: any, sourceArray: any[], destinationKey: any) => any[],
 }
 
-export interface Profile<TSourceKey extends string, TSource, TDestinationKey extends string, TDestination> {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type Profile<TSourceKey = string, TSource = any, TDestinationKey = string, TDestination = any> = {
   sourceKey: TSourceKey,
   destinationKey: TDestinationKey,
   map: (source: TSource, mapper: AnyMapper) => TDestination,
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type AnyProfile = Profile<string, any, string, any>
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type ExtractSourceKeys<TProfile> = TProfile extends Profile<infer TSourceKey, any, any, any> ? TSourceKey : never
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type ExtractSources<TProfile, TKey extends ExtractSourceKeys<TProfile>> = TProfile extends Profile<TKey, infer TSource, any, any> ? TSource : never
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type ExtractDestinationKeys<TProfile, TSourceKey extends ExtractSourceKeys<TProfile>> = TProfile extends Profile<TSourceKey, any, infer TDestinationKey, any> ? TDestinationKey : never
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type ExtractDestinations<TProfile, TSourceKey extends ExtractSourceKeys<TProfile>, TDestinationKey extends ExtractDestinationKeys<TProfile, TSourceKey>> = TProfile extends Profile<TSourceKey, any, TDestinationKey, infer TDestination> ? TDestination : never
+export type ProfileSource<TProfile extends Profile> = Parameters<TProfile['map']>[0]
```

Comment:
This is the only piece I'm not sure about. This specifically takes the first parameter. Which seems to have the same effect since in order to satisfy `Profile` the first parameter must be the source and the second can optionally be the mapper. But a hard coded `[0]` seems like a red flag here.